internal/report/merge.go

```diff
@@ -140,6 +141,59 @@ func combineCredits(creditSets ...[]models.Credit) []models.Credit {
 	return creditList
 }
 
+func rangeEventParse(r models.Range) (introduced, lastAffected, fixed string, limit []string) {
+	for _, e := range r.Events {
+		switch {
+		case e.Introduced != "":
+			introduced = e.Introduced
+		case e.Fixed != "":
+			fixed = e.Fixed
+		case e.LastAffected != "":
+			lastAffected = e.LastAffected
+		case e.Limit != "":
+			limit = append(limit, e.Limit)
+		}
+	}
+	if len(limit) == 0 {
+		limit = []string{"*"}
+	}
+	slices.Sort(limit)
+	return
+}
+
+func rangeEqual(r1, r2 models.Range) bool {
+	if !(r1.Type == r2.Type && r1.Repo == r2.Repo) {
+		// Basic details are not the same.
+		return false
+	}
+	intro1, lastAff1, fixed1, limit1 := rangeEventParse(r1)
+	intro2, lastAff2, fixed2, limit2 := rangeEventParse(r2)
+	if !(intro1 == intro2 && lastAff1 == lastAff2 && fixed1 == fixed2 && slices.Equal(limit1, limit2)) {
+		// Events are not the same.
+		return false
+	}
+	// Either we don't have any database specific entries, or they are entirely equal.
+	return (len(r1.DatabaseSpecific) == 0 && len(r2.DatabaseSpecific) == 0) || reflect.DeepEqual(r1, r2)
+}
+
+// combineRanges combines and reduces one or more range slices.
+//
+// The function assumes the input ranges are valid OSV ranges. If two or more
+// ranges are identical, only one will be included.
+func combineRanges(rangeSets ...[]models.Range) []models.Range {
```

Comment:
Would it make sense to add some logic to detect cases like 

```
{
  "introduced": "0"
}
```

And have this consume and remove any other ranges, given that `introduced: 0` on its own expresses every single version in existence is affected.

Reply:
I did consider doing that, but there are some considerations that led to me deferring the work.

For example, how do we handle merging two OSV where one is overly broad and uses `{"introduced": "0"}`, but the other is more specific and accurate?